kasa/smart/modules/firmware.py

```diff
@@ -116,3 +126,7 @@ async def set_auto_update_enabled(self, enabled: bool):
         """Change autoupdate setting."""
         data = {**self.data["get_auto_update_info"], "enable": enabled}
         await self.call("set_auto_update_info", data)  # {"enable": enabled})
+
+    async def _check_supported(self) -> bool:
+        """Check the module has auto_update or is connected."""
+        return self._device._is_cloud_connected or self.supported_version > 1
```

Comment:
If we'd create the feature always even when it's unavailable, we could avoid this and simplify the code further.

Reply:
Done